charts/gardener/controlplane/charts/runtime/templates/apiserver/deployment.yaml

```diff
@@ -277,6 +288,16 @@ spec:
         - name: gardener-apiserver-cert
           mountPath: /etc/gardener-apiserver/srv
           readOnly: true
+        {{- if .Values.global.apiserver.tlsSecretName }}
+        - name: gardener-apiserver-tls
+          mountPath: /etc/gardener-apiserver/srv/tls
+          readOnly: true
+        {{- end }}
+        {{- if .Values.global.apiserver.etcd.tlsSecretName }}
+        - name: gardener-etcd-client-tls
+          mountPath: /etc/gardener-apiserver/srv/etcd/
+          readOnly: true
+        {{- end }}
```

Comment:
If I understand your intention correctly, if both `.Values.global.apiserver.tlsSecretName` and `.Values.global.apiserver.etcd.tlsSecretName` are specified, then the `gardener-apiserver-cert` mount (here and below) should no longer be needed. In this case, it should rather not be present, as well as its corresponding checksum annotation. The corresponding secret could also not be generated at all.

Also, for the 2 new secrets that are mounted, checksum annotations should also be added to ensure that if they change, the deployment changes as well and its pods are restarted. The checksums would have to be generated by the code / script that applies the Helm charts (since the secrets are external), and then added via some additional values.